Subject: [PATCH] Hide unsupported stock OTA references on Android 17

Used only when the UN1CA updater is absent. Retain the separate GPSU,
Mainline and signed APEX update paths.

diff --git a/smali_classes3/com/samsung/android/settings/privacy/SecurityDashboardUtils.smali b/smali_classes3/com/samsung/android/settings/privacy/SecurityDashboardUtils.smali
--- a/smali_classes3/com/samsung/android/settings/privacy/SecurityDashboardUtils.smali
+++ b/smali_classes3/com/samsung/android/settings/privacy/SecurityDashboardUtils.smali
@@ -1509,54 +1509,11 @@
 .end method
 
 .method public static isSecurityUpdateSupported(Landroid/content/Context;)Z
-    .locals 4
-
-    invoke-static {}, Landroid/os/UserHandle;->myUserId()I
-
-    move-result v0
-
-    invoke-static {p0}, Lcom/samsung/android/settings/SecUserUtil;->getMainUserId(Landroid/content/Context;)I
-
-    move-result v1
-
-    const/4 v2, 0x0
-
-    if-ne v0, v1, :cond_0
-
-    const-string v0, "isOTAUpgradeAllowed"
-
-    const/4 v1, 0x0
-
-    const-string v3, "content://com.sec.knox.provider/RestrictionPolicy3"
-
-    invoke-static {p0, v3, v0, v1}, Lcom/android/settings/Utils;->getEnterprisePolicyEnabled(Landroid/content/Context;Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;)I
-
-    move-result p0
-
-    new-instance v0, Ljava/lang/StringBuilder;
-
-    const-string v1, "isOTAUpgradeAllowed : "
-
-    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v0
-
-    const-string v1, "PrintSettingsFragment"
-
-    invoke-static {v1, v0}, Landroid/util/secutil/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    if-eqz p0, :cond_0
-
-    const/4 p0, 0x1
-
-    return p0
-
-    :cond_0
-    return v2
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
 .method public static isTheftProtectionSupported(Landroid/content/Context;)Z
diff --git a/smali_classes3/com/samsung/android/settings/homepage/TopLevelSoftwareUpdatePreferenceController.smali b/smali_classes3/com/samsung/android/settings/homepage/TopLevelSoftwareUpdatePreferenceController.smali
--- a/smali_classes3/com/samsung/android/settings/homepage/TopLevelSoftwareUpdatePreferenceController.smali
+++ b/smali_classes3/com/samsung/android/settings/homepage/TopLevelSoftwareUpdatePreferenceController.smali
@@ -23,39 +23,11 @@
 .end method
 
 .method public getAvailabilityStatus()I
-    .locals 2
-
-    iget-object v0, p0, Lcom/android/settingslib/core/AbstractPreferenceController;->mContext:Landroid/content/Context;
-
-    const-string v1, "com.wssyncmldm"
-
-    invoke-static {v0, v1}, Lcom/android/settings/Utils;->isPackageEnabled(Landroid/content/Context;Ljava/lang/String;)Z
-
-    move-result v0
-
-    if-nez v0, :cond_0
-
-    const/4 p0, 0x3
-
-    return p0
-
-    :cond_0
-    iget-object p0, p0, Lcom/android/settingslib/core/AbstractPreferenceController;->mContext:Landroid/content/Context;
-
-    invoke-static {p0}, Lcom/samsung/android/settings/softwareupdate/SoftwareUpdateUtils;->needSoftwareUpdateBlockonDexMode(Landroid/content/Context;)Z
-
-    move-result p0
-
-    if-eqz p0, :cond_1
-
-    const/4 p0, 0x5
-
-    return p0
-
-    :cond_1
-    const/4 p0, 0x0
-
-    return p0
+    .locals 1
+
+    const/4 v0, 0x3
+
+    return v0
 .end method
 
 .method public bridge synthetic getBackgroundWorkerClass()Ljava/lang/Class;
